hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/InformixSqlAstTranslator.java

```diff
@@ -143,6 +144,18 @@ protected String getFromDualForSelectOnly() {
 		return " from " + getDual() + " dual";
 	}
 
+	@Override
+	protected void renderNull(Literal literal) {
+		if ( getParameterRenderingMode() == SqlAstNodeRenderingMode.NO_UNTYPED ) {
+			renderCasted( literal );
+		}
+		else {
+			int sqlType = literal.getExpressionType().getSingleJdbcMapping().getJdbcType().getJdbcTypeCode();
+			String nullString = getDialect().getSelectClauseNullString( sqlType, getSessionFactory().getTypeConfiguration() );
+			appendSql( nullString );
+		}
```

Comment:
Why not just this?

```suggestion
		renderCasted( literal );
```

Reply:
In this case there will be StackOverflowError with infinite loop:

	at org.hibernate.community.dialect.InformixSqlAstTranslator.renderNull(InformixSqlAstTranslator.java:149)
	at org.hibernate.sql.ast.spi.AbstractSqlAstTranslator.visitLiteral(AbstractSqlAstTranslator.java:7409)
	at org.hibernate.sql.ast.spi.AbstractSqlAstTranslator.visitQueryLiteral(AbstractSqlAstTranslator.java:7399)
	at org.hibernate.sql.ast.tree.expression.QueryLiteral.accept(QueryLiteral.java:54)
	at org.hibernate.sql.ast.spi.AbstractSqlAstTranslator.render(AbstractSqlAstTranslator.java:7073)
	at org.hibernate.query.sqm.produce.function.internal.PatternRenderer.render(PatternRenderer.java:224)
	at org.hibernate.query.sqm.produce.function.internal.PatternRenderer.render(PatternRenderer.java:159)
	at org.hibernate.query.sqm.produce.function.internal.PatternRenderer.render(PatternRenderer.java:142)
	at org.hibernate.dialect.function.CastFunction.render(CastFunction.java:76)
	at org.hibernate.sql.ast.spi.AbstractSqlAstTranslator.renderCasted(AbstractSqlAstTranslator.java:5813)
	at org.hibernate.community.dialect.InformixSqlAstTranslator.renderNull(InformixSqlAstTranslator.java:149)